apps-rendering/src/components/Callout/formFields.tsx

```diff
@@ -36,6 +37,18 @@ export const FormField: FC<FormFieldProp> = ({
 	const fieldValue =
 		formField.id in formData ? (formData[formField.id] as string) : '';
 	const fieldError = validationErrors[formField.id];
+	const firstUpdate = useRef(true);
+
+	useEffect(() => {
+		if (
+			firstUpdate.current &&
+			formField.hidden &&
+			formField.type === 'text'
+		) {
+			setFieldInFormData(formField.id, window.location.pathname);
```

Comment:
Do we do this on the client as we're unsure as to what the URL is on the server?

Reply:
This happens on the client only because `useEffect` doesn't work on SSR
https://codewithhugo.com/react-useeffect-ssr/